Bugs in DrILL interface

gui-co opened the pull request:
**Description of work.**

This PR solves several bugs in DrILL interface:
1. ILL data directory on Linux
2. remove of leading an trailing spaces in `CustomOptions`
3. automatic export when the default save directory setting is not set
4. automatic increment with special value 0

**To test:**

1. code review
2. try to write something like `Clear2DWorkspace=False ; ThetaDependent=False ` in the CustomOptions column in DrILL
3. clear the default save directory setting, process some data through DrILL and observe the logs
4. increment/decrement with 0 in the cell. Decrement from a number to see that 0 is never reached

Part of #30592 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

gvardany commented on scripts/Interface/ui/drill/model/DrillExportModel.py:
```diff
@@ -190,7 +190,10 @@ def run(self, sample):
         exportPath = config.getString("defaultsave.directory")
         if not exportPath:
-            exportPath = os.getcwd()
+            logger.warning("Default save directory is not defined. There will "
```
```suggestion
            logger.warning("Default save directory is not defined. Please specify one in the data directories dialog to enable exports.")
```

gui-co replied:
done

gvardany reviewed (commented):
I get the warning for each row.
Is it possible to check only once and show one warning?
Otherwise is fine.

gvardany commented:
An unrelated comment, the tooltip of CustomOptions header says CustomOptions, that's not very informative.
It should rather say: "Provide semicolon (;) separated key=value pairs"

gvardany commented:
Btw, it should be mentioned in the documentation, that there will be no export until one specifies a data save directory.

gui-co commented:
I changed the warning message and mentioned in the documentation that the export is cancelled if no save directory is provided. It will not be easy to warn only once as each row (sample) is treated independently. I can still modify it, let me know when you want to obtain the warning (when the processing is started?).

I also changed to tooltip of the `CustomOptions` column.

gvardany reviewed (approved):
Looks good, approving since urgent. The minor issue for when to check for the save directory can be added in a separate PR.

martyngigg commented:
My apologies. I tried to click "Enable auto-merge" and the page move around and I then clicked Close by accident.
I've enabled auto merge now so this should go in automatically when the builds finish
